File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/LocalDatabase.java
package com.vladislavmyasnikov.courseproject.data.db;

import com.vladislavmyasnikov.courseproject.data.db.converter.DateConverter;
import com.vladislavmyasnikov.courseproject.data.db.dao.LectureDao;
import com.vladislavmyasnikov.courseproject.data.db.dao.TaskDao;
import com.vladislavmyasnikov.courseproject.data.db.entity.LectureEntity;
import com.vladislavmyasnikov.courseproject.data.db.entity.TaskEntity;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {LectureEntity.class, TaskEntity.class}, version = 1, exportSchema = false)
@TypeConverters(DateConverter.class)
public abstract class LocalDatabase extends RoomDatabase {

    public abstract LectureDao lectureDao();
    public abstract TaskDao taskDao();
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/LocalDatabase.kt
package com.vladislavmyasnikov.courseproject.data.db

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.vladislavmyasnikov.courseproject.data.db.converter.DateConverter
import com.vladislavmyasnikov.courseproject.data.db.dao.*
import com.vladislavmyasnikov.courseproject.data.db.entities.*

@Database(entities = [LectureEntity::class, TaskEntity::class, StudentEntity::class, EventEntity::class], version = 1, exportSchema = false)
@TypeConverters(DateConverter::class)
abstract class LocalDatabase : RoomDatabase() {

    abstract fun lectureDao(): LectureDao
    abstract fun taskDao(): TaskDao
    abstract fun studentDao(): StudentDao
    abstract fun eventDao(): EventDao
}
